Azure/sonic-buildimage
# Build infrastructure changes to support sflow docker and utilities

**- What I did**

Add build infrastructure changes to support sflow. 

**- How I did it**

  Introduce a new "sflow" container (if ENABLE_SFLOW is set). The new docker will include:
	hsflowd : host-sflow based daemon is the sFlow agent
	psample : Built from libpsample repository. Useful in debugging sampled packets/groups.
	sflowtool : Locally dump sflow samples (e.g. with a in-unit collector)

  In case of SONiC-VS, enable psample & act_sample kernel modules.

  VS' syncd needs iproute2=4.20.0-2~bpo9+1 & libcap2-bin=1:2.25-1 to support tc-sample

  tc-syncd is provided as a convenience tool for debugging (e.g. tc-syncd filter show ...)

**- How to verify it**

1. Build sonic-vs.img.gz with and without ENABLE_SFLOW set
2. With sflow enabled, check the following in vs image:
    a. act_sample/psample
    b. docker exec -it sflow /bin/bash and inspect status
    c. docker stop/start sflow
    d. Verify sflowtool, psample and tc-syncd

**- Description for the changelog**

Add build infrastructure changes to support sflow. 

**- A picture of a cute animal (not mandatory but encouraged)**
